content/CustomExceptions/CustomExceptions.q.md

```diff
@@ -0,0 +1,1501 @@
+
+On line 7 of `main.php`, what does the statement `throw new DivideByZeroException("The value '0' as a divisor is not allowed.");` do?
+
+ - It enables the `divide()` function to throw an exception if an error occurs.
+
+ - It throws an exception if the condition `$divisor == 0` evaluates to `true`.
+
+ - It throws an exception if the condition `$divisor == 0` evaluates to `false`.
+
+ - It accesses the `Exception` class argument `The value '0' as a divisor is not allowed.`.
+
+ - It creates a new object of the `DivideByZeroException` class passing the argument `The value '0' as a divisor is not allowed.`.
+
+
+/// type=SS, id=56086bff-cef5-4a36-8d78-67177ad6ef77, answer=[5]
+
+On lines 13, 14, and 14 of `main.php`, what does the `try { echo divide(10, 0); }` block do?
+
+ - It evaluates the `divide(10, 0)` function call in the `echo` statement.
+
+ - It throws an exception when an error occurs in the `divide(10, 0)` function call.
+
+ - It returns an exception when an error occurs in the `divide(10, 0)` function call.
+
+ - It displays the exception message `The value '0' as a divisor is not allowed.`.
+
+ - It facilitates the catching of a possible exception thrown by the `divide(10, 0)` function call.
+
+
+/// type=MS, id=f3dd3528-b931-4f2b-a323-7d5e621ddce7, answer=[4,5]
+
+On lines 15, 16, and 17 of `main.php`, what does the `catch (DivideByZeroException $div) { echo "Exception caught: " . $div->errorMessage(); }` block do?
+
+ - It enables the `divide()` function to throw an exception if an error occurs.
+
+ - It throws an exception when an error occurs in `DivideByZeroException $div`.
+
+ - It returns an exception when an error occurs in `DivideByZeroException $div`.
+
+ - It catches the possible `DivideByZeroException` exception that occurs in the `try` block.
+
+ - It executes the `echo` statement if a `DivideByZeroException` exception occurs in the `try` block.
+
+
+/// type=MS, id=7bcd3acd-3915-4c46-95d7-aa22a5a3ab5f, answer=[2,3,5]
+
+Which statements correctly describe the `DivideByZeroException` class?
+
+ - It throws an exception when an error occurs in a program.
+
+ - It handles the possible exception that occurs in a program.
+
+ - It is a user-defined exception class that inherits the `Exception` class.
+
+ - It catches the possible `Exception` exception that occurs in a program.
+
+ - It has the `errorMessage()` method that returns an error message if an exception occurs.
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true, filename=[main.php,DivideByZeroException.php,DivideByNegativeException.php]
+
+```php
+<?php
+require_once(__DIR__ . "/DivideByZeroException.php");
+require_once(__DIR__ . "/DivideByNegativeException.php");
+
+function divide($dividend, $divisor)
+{
+    if ($divisor == 0) {
+        throw new DivideByZeroException("The value '0' as a divisor is not allowed.");
+    } elseif ($divisor < 0) {
+        throw new DivideByNegativeException();
+    } else {
+        return ($dividend / $divisor);
+    }
```

Comment:
simplify this also by having two `if` conditions and removing the `elseif` and `else`.